src/test/java/seedu/address/testutil/AllocateRoomDescriptorBuilder.java

```diff
@@ -27,7 +27,7 @@ public AllocateRoomDescriptorBuilder(Room room) {
         descriptor = new AllocateRoomDescriptor();
         descriptor.setRoomNumber(room.getRoomNumber());
         descriptor.setOccupied(room.isOccupied());
-        descriptor.setPatientName(room.getPatient().get().getName());
+        descriptor.setPatientName(room.getPatient().isPresent() ? room.getPatient().get().getName() : null);
```

Comment:
Perhaps we can simplify this expression?

`descriptor.setPatientName(room.getPatient().map(patient -> patient.getName()).orElse(null));`

Reply:
this will throw an exception because `patient` might be nulll. therefore, `patient.getName()` will give you null pointer exception

Reply:
nvm i tried it, it works, im a dumbzo HAHAHA